### ISSUE TITLE
[Bugs] Theming

### ISSUE THREAD

**Issue by bohning** [UltraStar-Deluxe/USDX#272]
The following is a list of issues that I have encountered while revising the Deluxe/Modern theme to support higher resolutions (goal was to support resolutions up to 3840x2160 (UHD)). I therefore tried my best to create scalable vector graphic files (using Inkscape) of all items and subsequently created high-resolution PNGs from them (5 times the size, as 3840/800 = 4.8). Currently, the sizes are hard-coded in the theme ini file to scale to an 800x600 screen and are therefore correctly downsized.

My first questions are general:
a. Should this fixed 800x600 scheme be reworked into something more flexible? Currently, the resizing to a 4:3 aspect ratio and then back to a higher, in many cases 16:9 aspect ratio resolution will result in distortions in the graphics if their sizes are fixed.
b. The ruler at the bottom shows (some!) navigation/shortcut keys, but the list is far from complete. E.g. 'Q' for quit is missing, 'P' for pause, 'E' for launching the editor, 'S' and 'D' for medleys... My suggestion is to move all keyboard shortcuts/navigation to a help system (accessible - hide and unhide overlay - via TAB) so that the skin itself is rather clean (the user learns the most obvious keys as arrow keys/Enter/Esc really quick anyways and will soon not need these anymore, yet these are the ones that are shown on basically every screen).

1. [interface]select_arrow_left.png and [interface]select_arrow_right.png
Their sizes are not determined in the theme ini file, thus they appear 5 times as large:
<img width="420" alt="bildschirmfoto 2017-04-22 um 11 58 09" src="https://cloud.githubusercontent.com/assets/1497707/25303475/02aceb4c-2754-11e7-9b2a-955e51e7edc8.png">

2. [sing]notes*.png
The same seems to be true for the drawing of the notes. There are gaps between the left cap, the middle part and the right cap, both for the background and the foreground (same is true for the editor, obviously):
<img width="420" alt="bildschirmfoto 2017-04-22 um 12 06 32" src="https://cloud.githubusercontent.com/assets/1497707/25303497/6f7f2672-2754-11e7-9363-beab7c2035dd.png">

3. [score]level*.png
Same issue here, while it is not the case for [score]endcap.png and [score]glass_box:
<img width="420" alt="bildschirmfoto 2017-04-22 um 12 09 17" src="https://cloud.githubusercontent.com/assets/1497707/25303513/ce7b10e6-2754-11e7-9ed4-b1cde7c76e1f.png">

4. [sing]pause.png
There is something strange going on with the Pause overlay, it seems to be overlayed a second time with distorted dimensions  (see duplicate "PA" on the right side of the screenshot):
<img width="420" alt="bildschirmfoto 2017-04-23 um 17 32 10" src="https://cloud.githubusercontent.com/assets/1497707/25314811/04a61a0c-284b-11e7-84ad-59a23933f205.png">

**Comment by bohning** [comment]
Intermediate status of the theme: https://www.dropbox.com/s/qauk9yl961wkg61/Edge.zip?dl=0

**Comment by bohning** [comment]
Another update, now with almost all graphics available as SVGs.
https://www.dropbox.com/s/qauk9yl961wkg61/Edge.zip?dl=0

**Comment by natrius** [comment]
Is this still somewhere available? There should be a download section for themes actually.

**Comment by basisbit** [comment]
there is no need for a "themes download section" here because all the themes which are maintained by the developers are already included with the game installer anyways.

**Comment by natrius** [comment]
Ah, it seems i downloaded a older version with some themes in a "Skins" folder. So i will stick with the 2 skins now available with the official download. 

Maybe this issue should be closed then @basisbit

**Comment by basisbit** [comment]
@natrius this issue here is about issues with the current theming system, not about your request, thus I think this issue should stay open :)

**Comment by natrius** [comment]
My bad, thought here was someone making a new theme for himself, instead he is reviewing the current themes so they work properly. Did not read as properly as i should before commenting. Thanks for answering!

**Comment by dgruss** [comment]
the two things are addressed by #1146 and https://github.com/UltraStar-Deluxe/Themes ?

**Comment by barbeque-squared** [comment]
reopened because I'm not sure if everything has been addressed yet, I _think_ the current status is:
* a: this is a long-term goal, the code isn't ready for this yet
* b: probably not fixed yet, though not yet sure how we want to approach this either (I agree what's there currently is a bit of a mess, but there's multiple ways to approach this, might eventually warrant its own issue)
* 1: should be fixed since #1213
* 2: I don't know
* 3: I don't know
* 4: I don't know

If someone knows the status of 2, 3 and 4 I'll edit the description and do some strikethroughs or something

**Comment by dgruss** [comment]
All are fixed:

## **1:** fixed by #1213. Immediately before #1213, the arrows are rendered at their enlarged source size:

<img width="1600" height="900" alt="Image" src="https://github.com/user-attachments/assets/fb3d6a4d-dcd3-4ab4-9144-ae9a1615af85" />

Current master respects the dimensions configured by the theme:

<img width="1600" height="900" alt="Image" src="https://github.com/user-attachments/assets/614e3cdc-1d87-406b-841e-0c808cc9fff8" />

## **2:** fixed as well (issue is from 2017!) as I cannot reproduce the reported gaps. The 5x note textures render without visible gaps both while singing and in the editor:

<img width="1600" height="900" alt="Image" src="https://github.com/user-attachments/assets/1a2231db-bc9c-4655-beb9-67251308936e" />

<img width="1600" height="900" alt="Image" src="https://github.com/user-attachments/assets/5d6ab87d-5de9-45c0-bb4f-d27e97169f7b" />

## **3:** injected a 10,000-point score so that all three `level_*` textures were visible. The 5× textures remain correctly sized, stacked and contained inside the score bar:

<img width="1600" height="900" alt="Image" src="https://github.com/user-attachments/assets/dd9208ec-627d-4c50-92dd-ba744866a5e5" />

## * **4:** a 4000×3000 pause texture produces one correctly sized pause overlay, without the duplicated/distorted second `PAU...`:

<img width="1600" height="900" alt="Image" src="https://github.com/user-attachments/assets/6830ccdf-70f9-4d8c-b557-a35694199a28" />

So, this can be closed as completed

**Comment by barbeque-squared** [comment]
Yeah I'm going to close this then. The 800x600 thing is on the radar anyway (I don't need a ticket to remind me of that) and what shortcuts we show we can definitely have a discussion on, but it's not really a bug and probably also useful to have a dedicated ticket to that in the future.
